### ISSUE TITLE
Internal Server Error (File "/app/calibre-web/cps/db.py", line 692, in fill_indexpage_with_archived_books)

### ISSUE THREAD

**Issue by jeff47** [janeczku/calibre-web#2185]
**Describe the bug/problem**
Getting an error about an internal webserver error, possibly in File db.py?

**To Reproduce**
Have tried removing the container, pulling a new one.  No change.  It was working until a few days ago, I'm not sure what changed.

**Logfile**
Here's the output in the brower:
```
Calibre-Web
Internal Server Error
500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
Traceback (most recent call last):
File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 2073, in wsgi_app
response = self.full_dispatch_request()
File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1518, in full_dispatch_request
rv = self.handle_user_exception(e)
File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1516, in full_dispatch_request
rv = self.dispatch_request()
File "/usr/local/lib/python3.8/dist-packages/flask/app.py", line 1502, in dispatch_request
return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
File "/app/calibre-web/cps/usermanagement.py", line 38, in decorated_view
return login_required(func)(*args, **kwargs)
File "/usr/local/lib/python3.8/dist-packages/flask_login/utils.py", line 272, in decorated_view
return func(*args, **kwargs)
File "/app/calibre-web/cps/web.py", line 744, in index
return render_books_list("newest", sort_param, 1, page)
File "/app/calibre-web/cps/web.py", line 408, in render_books_list
entries, random, pagination = calibre_db.fill_indexpage(page, 0, db.Books, True, order,
File "/app/calibre-web/cps/db.py", line 679, in fill_indexpage
return self.fill_indexpage_with_archived_books(page, pagesize, database, db_filter, order, False, *join)
File "/app/calibre-web/cps/db.py", line 692, in fill_indexpage_with_archived_books
query = self.session.query(database)
AttributeError: 'NoneType' object has no attribute 'query'
```

I think this part from from docker logs might be relevant:  (full output is attached)
```
Reading state information...
The following NEW packages will be installed:
  libxdamage1
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 6,996 B of archives.
After this operation, 31.7 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu focal/main amd64 libxdamage1 amd64 1:1.1.5-2 [6,996 B]
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76, <> line 1.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.30.0 /usr/local/share/perl/5.30.0 /usr/lib/x86_64-linux-gnu/perl5/5.30 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.30 /usr/share/perl/5.30 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7, <> line 1.)
debconf: falling back to frontend: Teletype
dpkg-preconfigure: unable to re-open stdin: 
```
Output from docker logs calibre-web:
[docker-calibre-web.log](https://github.com/janeczku/calibre-web/files/7582820/docker-calibre-web.log)

Calibre-web log:
[calibre-web.log](https://github.com/janeczku/calibre-web/files/7582819/calibre-web.log)

**Environ
ment (please complete the following information):**
 - OS: linux
 - Python version: [e.g. python2.7]
 - Calibre-Web version: [e.g. 0.6.8 or 087c4c59 (git rev-parse --short HEAD)]:
 - Docker container: linuxserver
 - Special Hardware: no
 - Browser: Brave

**Additional context**
Happy to provide any additional logs etc.

**Comment by jeff47** [comment]
I'm going to venture a guess that this is similar to #[2184](https://github.com/janeczku/calibre-web/issues/2184) and #[2173](https://github.com/janeczku/calibre-web/issues/2173).  The way the error comes up in the browser with a link to github issues gives the impression that a complete bug report will be created automatically...

**Comment by jeff47** [comment]
Seemed to be resolved by removing app.db, although that's not optimal since it resets all settings.

**Comment by OzzieIsaacs** [comment]
You could send me the app.db file and I‘ll have a look at it

**Comment by jeff47** [comment]
Thanks - I'm not convinced that was the issue.  I've got it working again and I'm not clear what is different.  I downgraded the version to see if any recent changes were at issue, but same error with 0.6.13.  So then I moved app.db and gdrive.db to a backup, and it worked.  

Then I updated it to the most recent version, and it still worked.  So to confirm it was app.db, I copied that back from my backup and... it still works.  

So I've got nothing that can help.  I guess close this and I'll touch base if it happens again.
